src/eureka/S3_data_reduction/nirspec.py

```diff
@@ -155,3 +155,201 @@ def fit_bg(dataim, datamask, n, meta, isplots=0):
         The current integration number.
     """
     return nircam.fit_bg(dataim, datamask, n, meta, isplots=isplots)
+
+
+def find_column_coms_and_shifts(data):
+    '''
+    Takes a data array (int, rows, columns) and find the 
+    center of mass (COM) in pixels for each column. It then returns the needed
+    shift to apply to each column to bring the COM to the center
+
+    Input:
+    data : (array) sequence of images. Shape: (Nb_integrations x Nb_rows x Nb_columns)
+
+    Return:
+    shifts : (array) shift to apply to each column of each integration to move
+                the trace to the middle of the detector and make it straight
+                Shape: (Nb_integrations x Nb_columns)
+
+
+    '''
+
+    # get the dimensions of the data
+    nb_rows = data.shape[1]
+
+    # define a array of pixel positions (in their centers)
+    pix_centers = np.arange(nb_rows) + 0.5
+
+    # Compute the center of mass of each column and convert to integer (pixels)
+    column_coms = np.sum(pix_centers[None, :, None]*data, axis=1) / np.sum(data, axis=1)
+    column_coms = np.around(column_coms).astype(int)
+
+    # # try using the max of each column
+    # column_max = np.argmax(data, axis=1)
+    # column_coms = column_max.astype(int)
+
+    # define the new center (where we will align the trace) in the middle of the detector
+    new_center = int(nb_rows/2)
+
+    # define an array containing the needed shift to bring the COMs to the
+    # new center for each column of each integration
+    shifts = new_center - column_coms
+
+    # ensure columns of zeros or nans are not moved and dont induce bugs
+    shifts[column_coms < 0] = 0
+    shifts[column_coms > nb_rows] = 0
+
+    return shifts, new_center
+
+
+def find_column_median_shifts(data):
```

Comment:
This function is extremely similar to that above with lots of duplicated code - can you just add something like the following to the top of function above to avoid two functions that do basically the same thing?
```
if len(data.shape) == 2:
    # Dealing with the median frame, so reshape
    data = data[np.newaxis, :, :]
```

Reply:
One of the function was unused for now, so I simply removed it and kept the one that is used by the code.